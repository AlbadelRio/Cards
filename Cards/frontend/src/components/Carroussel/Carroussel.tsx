import * as React from 'react';
import { useState } from 'react';

import {
  FlatList,
  StatusBar,
  View,
  Text,
  Pressable
} from 'react-native';

import styles from './carrousselStyle';

export default function Carroussel({ route: { params } }:any) {
  const { pack } = params;

  const [showAnswer, setShowAnswer] = useState(false);
  function flipHandler() {
    if (showAnswer) {
      setShowAnswer(false);
    } else {
      setShowAnswer(true);
    }
  }

  return (

    <View style={{ flex: 1, backgroundColor: '#5EBBB0' }}>
      <StatusBar hidden />
      <FlatList
        data={pack?.packCards}
        horizontal
        pagingEnabled
        renderItem={({ item, index }) => (
          <Pressable
            style={styles.card}
            key={index.toString()}
            onPress={flipHandler}
          >
            <View style={styles.inside}>
              { !showAnswer && (
              <>
                <Text style={styles.text}>Question</Text>
                <Text style={styles.text}>{item?.question}</Text>
              </>
              )}
              {showAnswer && (
              <>
                <Text style={styles.text}>Answer</Text>
                <Text style={styles.text}>{item?.answer}</Text>
              </>
              )}
              <Text style={styles.text}>Tap to Flip</Text>
            </View>

          </Pressable>

        )}
      />
    </View>
  );
}
