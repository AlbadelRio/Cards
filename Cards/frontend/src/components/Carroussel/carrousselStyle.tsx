import { StyleSheet, Dimensions } from 'react-native';

const { width } = Dimensions.get('screen');
const cardW = width * 0.7;
const cardH = cardW * 1.54;
export default StyleSheet.create({
  card: {
    width,
    justifyContent: 'center',
    alignItems: 'center',
    elevation: 30
  },
  inside: {

    backgroundColor: '#272727',
    width: cardW,
    height: cardH,
    borderRadius: 16,
    elevation: 2,
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: 8,
    textAlign: 'center'

  },
  text: {
    color: 'white',
    fontSize: 20,
    fontWeight: 'bold'
  }
});
