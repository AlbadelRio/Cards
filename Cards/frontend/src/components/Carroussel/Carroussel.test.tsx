import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import Carroussel from './Carroussel';

describe('when invoked a Carroussel component', () => {
  const route = {
    params: {
      pack: {
        id: '1234',
        packCards: [{ answer: 'answer' }]
      }
    }
  };
  test('Should render Carroussel component', () => {
    const carroussel = render(<Carroussel route={route} />);
    expect(carroussel).toMatchSnapshot();
  });

  test('when pressable 0, the flipHandler has to been called', () => {
    const screen = render(<Carroussel route={route} />);

    const pressableItem = screen.getByTestId('pressable0');
    fireEvent.press(pressableItem);
    expect(screen.queryByTestId('answer')).not.toBe(null);
  });
  describe('when is pressed twice', () => {
    test('then the answer should to be null', () => {
      const screen = render(<Carroussel route={route} />);

      const pressableItem = screen.getByTestId('pressable0');
      fireEvent.press(pressableItem);
      fireEvent.press(pressableItem);
      expect(screen.queryByTestId('answer')).toBeNull();
    });
  });
});
