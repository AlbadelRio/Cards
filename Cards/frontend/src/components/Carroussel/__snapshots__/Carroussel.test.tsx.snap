// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when invoked a Carroussel component Should render Carroussel component 1`] = `
<BVLinearGradient
  colors={
    Array [
      4284398512,
      4284309487,
    ]
  }
  endPoint={
    Object {
      "x": 0,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "answer": "answer",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "elevation": 30,
                "justifyContent": "center",
                "width": 750,
              }
            }
            testID="pressable0"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#272727",
                  "borderRadius": 16,
                  "elevation": 2,
                  "height": 808.5,
                  "justifyContent": "space-between",
                  "padding": 8,
                  "textAlign": "center",
                  "width": 525,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Question
              </Text>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Tap to Flip
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</BVLinearGradient>
`;
